src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -93,12 +101,48 @@ export abstract class VerifiableCredentialsServiceBase {
     proof_options: string
   ): Promise<string>;
 
-  constructor(protected readonly _signerService: SignerService) {}
+  constructor(
+    protected readonly _signerService: SignerService,
+    private readonly _storage: VCStorageClient
+  ) {}
+
+  /**
+   * @description requests verifiable credentials of type `type`. Only EWF credentials currently supported
+   * @returns credentials query with matching verifiable presentations
+   */
+  async initiateExchange({ type, url }: ExchangeInvitation) {
+    if (type !== EWF_EXCHANGE_INVITATION) {
+      throw new Error('Only EWF exchange is supported');
+    }
+
+    const { data: vpRequest } = await axios.post<VpRequest>(url);
+    const presentationDefinitions = vpRequest.query.find(
+      (q) => q.type === VpRequestQueryType.presentationDefinition
+    )?.credentialQuery as PresentationDefinition[];
+
+    const vc: VerifiableCredential<RoleCredentialSubject>[] = [];
+    const credentials = (
+      await Promise.all(
+        presentationDefinitions.map((def) =>
+          this._storage.getCredentialsByDefinition(def)
+        )
+      )
+    ).reduce((prev, curr) => prev.concat(curr), []);
+    for (const cred of credentials) {
+      if (isRoleCredential(cred)) {
+        vc.push(cred);
+      }
+    }
+    // }
+    const vp = await this.createVerifiablePresentation(vc);
+    return { vp, vpRequest };
+  }
 
   // * Should be overridden by the implementation
   static async create(
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    signerService: SignerService
+    signerService: SignerService,
+    storage: VCStorageClient
```

Comment:
```suggestion
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    storage: VCStorageClient
```

Reply:
I think we can delete this method at all. Why do we need it?

Reply:
We can create a static abstract method, and this is making sure that derived class has implemented this method

Reply:
In ES6 it is possible

Reply:
I've already done it. It will throw an error on runtime if derived class doesn't have this method